src/k8s-engine/lima.ts

```diff
@@ -1338,13 +1385,17 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         }
 
         // We have no good estimate for the rest of the steps, go indeterminate.
-        timers.clearInterval(this.progressInterval);
-        this.progressInterval = undefined;
+        if (enabledK3s) {
+          timers.clearInterval(this.progressInterval as ReturnType<typeof timers.setInterval>);
+          this.progressInterval = undefined;
+        }
 
         if ((await this.status)?.status === 'Running') {
           this.lastCommandComment = 'Stopping existing instance';
           await this.progressTracker.action(this.lastCommandComment, 100, async() => {
-            await this.ssh('sudo', '/sbin/rc-service', 'k3s', 'stop');
+            if (this.#enabledK3s) {
+              await this.ssh('sudo', '/sbin/rc-service', '--ifstarted', 'k3s', 'stop');
```

Comment:
If we always write out the initrd files, then we can always stop the `k3s` service safely without checking for `this.#enabledK3s`.